app/providers/search/duckduckgo.py

```
@@ -0,0 +1,56 @@
+        current_page = 1
+
+        while len(page_results_list) < start + num_results:
+            try:
+                page_data = ddgs.text(
+                    query=q,
+                    region=region,
+                    safesearch=safesearch,
+                    page=current_page,
+                    backend="auto"
+                )
+            except Exception:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)